pkg/handlers/v1/cloud_fetch.go

```diff
@@ -117,44 +164,113 @@ func (h *CloudFetchByHostnameHandler) Handle(ctx context.Context, input CloudAss
+// Handle handles fetching cloud assets with pagination
+func (h *CloudFetchAllAssetsByTimeHandler) Handle(ctx context.Context, input CloudAssetFetchAllByTimestampParameters) (PagedCloudAssets, error) {
 	logger := h.LogFn(ctx)
 
 	ts, e := time.Parse(time.RFC3339Nano, input.Timestamp)
 	if e != nil {
 		logger.Info(logs.InvalidInput{Reason: e.Error()})
-		return CloudAssets{}, InvalidInput{Field: "time", Cause: e}
+		return PagedCloudAssets{}, InvalidInput{Field: "time", Cause: e}
 	}
 
-	if input.Count == nil {
+	if input.Count == 0 {
 		e = errors.New("missing or malformed required parameter count")
 		logger.Info(logs.InvalidInput{Reason: e.Error()})
-		return CloudAssets{}, InvalidInput{Field: "count", Cause: e}
+		return PagedCloudAssets{}, InvalidInput{Field: "count", Cause: e}
 	}
 
-	var offset uint
-	if input.Offset != nil {
-		offset = *input.Offset
+	assetType, e := validateAssetType(input.Type)
+	if e != nil {
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
+		return PagedCloudAssets{}, InvalidInput{Field: "type", Cause: e}
 	}
 
-	assets, e := h.Fetcher.FetchAll(ctx, ts, *input.Count, offset)
+	var offset uint // do not offset as this is the first page
+	assets, e := h.Fetcher.FetchAll(ctx, ts, input.Count, offset, assetType)
 	if e != nil {
 		logger.Error(logs.StorageError{Reason: e.Error()})
-		return CloudAssets{}, e
+		return PagedCloudAssets{}, e
 	}
 	if len(assets) == 0 {
-		return CloudAssets{}, NotFound{ID: "any"}
+		return PagedCloudAssets{}, NotFound{ID: "any"}
 	}
 
-	return extractOutput(assets), nil
+	nextPageToken, e := input.toNextPageToken()
+	if e != nil {
+		logger.Error(logs.StorageError{Reason: e.Error()})
+	}
+	return PagedCloudAssets{extractOutput(assets), nextPageToken}, nil
+}
+
+// CloudFetchAllAssetsByTimePageHandler defines a lambda handler for bulk fetching subsequent pages of cloud assets known at specific point in time
+type CloudFetchAllAssetsByTimePageHandler struct {
+	LogFn   domain.LogFn
+	StatFn  domain.StatFn
+	Fetcher domain.CloudAllAssetsByTimeFetcher
+}
+
+// Handle handles subsequent page fetching of cloud assets
+func (h *CloudFetchAllAssetsByTimePageHandler) Handle(ctx context.Context, input CloudAssetFetchAllByTimeStampPageParameters) (PagedCloudAssets, error) {
+	logger := h.LogFn(ctx)
+	params, e := fetchAllByTimeStampParametersForToken(input.PageToken)
+	if e != nil {
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
+		return PagedCloudAssets{}, InvalidInput{Field: "pageToken", Cause: e}
+	}
+	ts, e := time.Parse(time.RFC3339Nano, params.Timestamp)
+	if e != nil {
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
+		return PagedCloudAssets{}, InvalidInput{Field: "pageToken", Cause: e}
+	}
+
+	//generic error to report to caller to avoid exposing the internal token structure NB, the specific error is still logged
+	tokenError := errors.New("malformed pageToken")
```

Comment:
nit: This could be a package constant/variable and re-used rather than redefined on each call.

Reply:
Great point